source/extensions/filters/network/common/redis/client.h

```diff
@@ -260,9 +258,14 @@ struct Transaction {
   bool active_{false};
   bool connection_established_{false};
   bool should_close_{false};
+
+  // The key which represents the transaction hash slot.
   std::string key_;
-  ClientPtr client_;
+  // clients_[0] represents the main connection, clients_[1..n] are for
+  // the mirroring policies.
+  std::vector<ClientPtr> clients_;
   Network::ConnectionCallbacks* connection_cb_;
+  uint32_t current_client_idx{0};
```

Comment:
The requests themselves are sent synchronously and in order, the replies may be received out of order but that's ok because each reply is mapped to the proper client connection.

Reply:
I've added the underscore + a comment on the variable name